sqlobject/src/main/java/org/jdbi/v3/sqlobject/SqlObjectFactory.java

```diff
@@ -105,38 +108,55 @@ public boolean accepts(Class<?> extensionType) {
 
     private Map<Method, Handler> buildHandlersFor(Class<?> sqlObjectType) {
         return handlersCache.computeIfAbsent(sqlObjectType, type -> {
-
             final Map<Method, Handler> handlers = new HashMap<>();
-            for (Method method : sqlObjectType.getMethods()) {
-                Optional<? extends Class<? extends HandlerFactory>> factoryClass = Stream.of(method.getAnnotations())
-                        .map(a -> a.annotationType().getAnnotation(SqlMethodAnnotation.class))
-                        .filter(Objects::nonNull)
-                        .map(a -> a.value())
-                        .findFirst();
-
-                if (factoryClass.isPresent()) {
-                    HandlerFactory factory = buildFactory(factoryClass.get());
-                    Handler handler = factory.buildHandler(sqlObjectType, method);
-                    handlers.put(method, handler);
-                }
-                else if (mixinHandlers.containsKey(method)) {
-                    handlers.put(method, mixinHandlers.get(method));
-                }
-                else {
-                    handlers.put(method, new PassThroughHandler());
-                }
-            }
 
             handlers.putAll(EqualsHandler.handler());
             handlers.putAll(ToStringHandler.handler(sqlObjectType.getName()));
             handlers.putAll(HashCodeHandler.handler());
             handlers.putAll(FinalizeHandler.handlerFor(sqlObjectType));
 
+            for (Method method : sqlObjectType.getMethods()) {
+                handlers.computeIfAbsent(method, m -> buildMethodHandler(sqlObjectType, m));
+            }
+
             return handlers;
         });
     }
 
-    private HandlerFactory buildFactory(Class<? extends HandlerFactory> factoryClazz) {
+    private Handler buildMethodHandler(Class<?> sqlObjectType, Method method) {
+        if (mixinHandlers.containsKey(method)) {
+            return mixinHandlers.get(method);
+        }
+
+        Handler handler = buildBaseHandler(sqlObjectType, method);
+        return addDecorators(handler, sqlObjectType, method);
+    }
+
+    private Handler buildBaseHandler(Class<?> sqlObjectType, Method method) {
+        return Stream.of(method.getAnnotations())
+                .map(a -> a.annotationType().getAnnotation(SqlMethodAnnotation.class))
+                .filter(Objects::nonNull)
+                .map(a -> buildFactory(a.value()))
+                .map(factory -> factory.buildHandler(sqlObjectType, method))
+                .findFirst()
```

Comment:
I worry that this depends on the iteration order of annotations on your method, which is inherently undefined.  Should we impose an ordering here, or fail if there are multiple annotations matching?